tools/rpmkeys.c

```diff
@@ -73,9 +72,30 @@ int main(int argc, char *argv[])
 	    rpmtsSetFlags(ts, (rpmtsFlags(ts)|RPMTRANS_FLAG_TEST));
 	ec = rpmcliImportPubkeys(ts, args);
 	break;
-    /* XXX TODO: actually implement these... */
     case MODE_DELKEY:
+	struct rpmInstallArguments_s * ia = &rpmIArgs;
+	ARGV_t gpgargs = argvNew();
+	for (char * const * arg = args; *arg; arg++) {
+	    if (strncmp(*arg, "gpg-pubkey-", 11)) {
+		char * gpgarg = rpmExpand("gpg-pubkey-", *arg, NULL);
+		argvAdd(&gpgargs, gpgarg);
+		free(gpgarg);
+	    } else {
+		argvAdd(&gpgargs, *arg);
+	    }
+	}
+	ec = rpmErase(ts, ia, gpgargs);
+	argvFree(gpgargs);
+	break;
     case MODE_LISTKEY:
+	if (args != NULL) {
+	    argerror(_("--list-keys does not take any arguments"));
+	}
+	ARGV_t query = argvSplitString("gpg-pubkey", " ", ARGV_NONE);
```

Comment:
It's a bit creative way to initialize an argv... I think
```
ARGV_t query = NULL;
argvAdd(&query, "gpg-pubkey");
```
...would be a well worth the one extra line in obviousness :smile: